config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -1209,6 +1258,7 @@ public CsrfSpec csrfTokenRepository(
 		public CsrfSpec requireCsrfProtectionMatcher(
 			ServerWebExchangeMatcher requireCsrfProtectionMatcher) {
 			this.filter.setRequireCsrfProtectionMatcher(requireCsrfProtectionMatcher);
+			this.specifiedRequireCsrfProtectionMatcher = true;
```

Comment:
I'm not a big fan of this as I'd prefer it be atomic, but I don't really see another way to tell whether or not the user called this method or not.

I tried introducing a new field to temporarily hold the matcher, and then defer the filter configuration to the `configure` method, but this presents the problem that once `JwtSpec` wants to configure `csrf`, `CsrfSpec#configure` has already been invoked.

This approach at least has the elegance that other Specs can continue to configure the filter after `CsrfSpec#configure` has been invoked.

Open to other ideas, of course.

Reply:
I think this approach is fine. There really isn't a way to avoid adding some sort of state to determine if the default is being used. The state is private, so we can always refactor later too.